conans/client/tools/scm.py

```diff
@@ -346,7 +346,14 @@ def get_branch(self):
             raise ConanException("Unable to get svn branch from %s: %s" % (self.folder, str(e)))
 
     def _check_svn_repo(self):
+        """ Check if it is a valid SVN repo """
         try:
-            self.run("info")
-        except Exception:
-            raise ConanException("Not a valid SVN repository")
\ No newline at end of file
+            with chdir(self.folder) if self.folder else no_op():
+                process = subprocess.Popen(['svn', 'info'],
+                                           stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+                process.communicate()
+        except FileNotFoundError:
+            raise ConanException("Not a valid SVN repository")
```

Comment:
probably it indicates `svn.exe` is missing, not an invalid repository?

Reply:
If SVN is missing it will fail too, but the intention of this function `_check_svn_repo` is to check if there is a repository where this command is being run (check where the function is being called).

Nevertheless, we should check what is happening when a recipe with SCM is being built and there is no SVN as you say, it will fail what we should give a meaningful exception.

Reply:
So the Popen raises a `FileNotFoundError` when there is not a valid svn repository? How? :open_mouth:
Is it not related wether the `self.folder` exists or not, right?

Reply:
The `self.folder` _always_ exists, look at the `__init__` for this class:

```python

class SCMBase(object):
    cmd_command = None

    def __init__(self, folder=None, verify_ssl=True, username=None, password=None,
                 force_english=True, runner=None, output=None):
        self.folder = folder or os.getcwd()
        if not os.path.exists(self.folder):
            os.makedirs(self.folder)

```

----

And the `Popen` raises a `FileNotFoundError` if the command does not exists, otherwise it is raising because of the `if bool(process.returncode)`.

Reply:
`command does not exists` means `svn.exe` doesn't exist, right? then raised exception message doesn't match the actual error happened and might be confusing for user.

Reply:
That's also my point.

Reply:
Maybe you could check if the svn interpreter is available by calling `svn --version --quiet` before

Reply:
Changed! 👍

Reply:
@ahauan4, this is an internal function to the class where it is declared, it was the reason I didn0t care about `svn` not being available, but I'm fine changing the error message.

If SVN (or Git) are not available, the SCM feature will fail and be should provide a meaningful error for both cases, and I think that circumstance is not covered yet.